duckdb-vortex/src/vortex_extension.cpp

```diff
@@ -64,6 +66,55 @@ static void VortexScanFunction(ClientContext &context, TableFunctionInput &data,
     state.finished = true;
 }
 
+/// Converts a Vortex data type to a DuckDB logical type.
+static LogicalType VortexTypeToDuckDBType(uint8_t dtype_tag) {
+    static const std::unordered_map<uint8_t, LogicalType> type_map = {
+        {DTYPE_BOOL, LogicalType::BOOLEAN},
+        {DTYPE_PRIMITIVE_I8, LogicalType::INTEGER},
+        {DTYPE_PRIMITIVE_I16, LogicalType::INTEGER},
+        {DTYPE_PRIMITIVE_I32, LogicalType::INTEGER},
+        {DTYPE_PRIMITIVE_I64, LogicalType::BIGINT},
+        {DTYPE_PRIMITIVE_U8, LogicalType::UINTEGER},
+        {DTYPE_PRIMITIVE_U16, LogicalType::UINTEGER},
+        {DTYPE_PRIMITIVE_U32, LogicalType::UINTEGER},
+        {DTYPE_PRIMITIVE_U64, LogicalType::UBIGINT},
+        {DTYPE_PRIMITIVE_F32, LogicalType::FLOAT},
+        {DTYPE_PRIMITIVE_F64, LogicalType::DOUBLE},
+        {DTYPE_UTF8, LogicalType::VARCHAR},
+        {DTYPE_BINARY, LogicalType::BLOB},
+    };
+
+    auto it = type_map.find(dtype_tag);
+    if (it != type_map.end()) {
+        return it->second;
+    }
+
+    // For unsupported types, default to VARCHAR.
+    return LogicalType::VARCHAR;
+}
+
+/// Extracts schema information from a Vortex file's data type.
+static void ExtractVortexSchema(const DType* file_dtype,
+                                vector<LogicalType>& column_types,
+                                vector<string>& column_names) {
+    uint32_t field_count = DType_field_count(file_dtype);
+    for (uint32_t idx = 0; idx < field_count; idx++) {
+        char name_buffer[512];
+        int name_len = 0;
```

Comment:
are you concerned about buffer overflow?

Reply:
make the buf bigger?